packages/zowe-explorer/src/utils/CertificateWizard.ts

```diff
@@ -0,0 +1,126 @@
+})();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+export type CertWizardOpts = {
+    cert?: string;
+    certKey?: string;
+    dialogOpts?: OpenDialogOptions;
+};
+
+class DeferredPromise<T> {
+    public promise: Promise<T>;
+    public resolve: (value: T | PromiseLike<T>) => void;
+    public reject: (reason?: any) => void;
+
+    public constructor() {
+        this.promise = new Promise<T>((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+        });
+    }
+}
+
+const allFiles = localize("fileDialog.allFiles", "All Files");
+const userDismissed = localize("zowe.certWizard.dismissed", "User dismissed the certificate wizard.");
+
+export class CertificateWizard extends WebView {
+    private opts: CertWizardOpts;
+    public userSubmission: DeferredPromise<{
+        cert: string;
+        certKey: string;
+    }> = new DeferredPromise();
+
+    public constructor(context: ExtensionContext, opts: CertWizardOpts) {
+        super(localize("cert.viewTitle", "Certificate Wizard"), "certificate-wizard", context, (message: object) =>
+            this.onDidReceiveMessage(message)
+        );
+        this.opts = opts;
+        this.panel.onDidDispose(() => {
+            this.userSubmission.reject(userDismissed);
+        });
+    }
+
+    protected async onDidReceiveMessage(message: any): Promise<void> {
+        switch (message.command) {
+            case "promptCert":
+                {
+                    const tempCert = await Gui.showOpenDialog({
+                        title: "Enter the path to the certificate for authenticating the connection.",
+                        defaultUri: this.opts.cert ? Uri.file(this.opts.cert) : undefined,
+                        filters: { [localize("cert.files", "Certificate Files")]: ["cer", "crt", "pem"], [allFiles]: ["*"] },
+                        openLabel: localize("cert.select", "Select Certificate"),
+                        ...(this.opts.dialogOpts ?? {}),
+                    });
+                    if (tempCert != null && tempCert[0] != null) {
+                        this.opts.cert = tempCert[0].fsPath;
+                        await this.panel.webview.postMessage({
+                            opts: {
+                                cert: tempCert[0].fsPath,
+                            },
+                        });
+                    }
+                }
+                break;
+            case "promptCertKey":
+                {
+                    const tempCertKey = await Gui.showOpenDialog({
+                        title: localize("cert.enterKeyPath", "Enter the path to the certificate key for authenticating the connection."),
+                        defaultUri: this.opts.certKey ? Uri.file(this.opts.certKey) : undefined,
+                        filters: {
+                            [localize("cert.keys", "Certificate Keys")]: ["cer", "crt", "pem", "key"],
+                            [allFiles]: ["*"],
```

Comment:
I'm still getting the debug-console error and no notification 😢 
- `rejected promise not handled within 1 second: Error: error:0480006C:PEM routines::no start line`

I'm testing over Remote SSH on a linux environment with dummy files (like zowe.config.json), so maybe it might be just me 😋

Reply:
I might have misinterpreted the consensus from our last stand-up discussion, but  I believe we'll need to handle this case within Imperative. Once the API is updated to throw an error for an invalid cert (rather than not handling the exception), the new messaging will apply to this feature 😋